### PR TITLE
Pilot integration tests - Closes #733

### LINKED ISSUE #733: Pilot integration tests
### Expected behaviour
There are many blog posts out there on how to do integration tests with Mocha and Enzyme
https://engineering.classdojo.com/blog/2017/01/12/integration-testing-react-redux/
https://medium.freecodecamp.org/real-integration-tests-with-react-redux-and-react-router-417125212638

We should try to write some integration tests to evaluate if they offer some benefits. Corrently I can think of these:
- Can replace some of the e2e tests while being much faster. IMO we still need e2e test for each feature, but maybe not for the edge/error cases, because our e2e suite takes about 4 minutes and is growing with every new feature.
- Can mock API responses to test API error cases which we cannot test in e2e tests, where we cannot make the server to return an error when we want.

### Actual behaviour
We use unit tests and e2e tests, but no integration tests.

### PR DESCRIPTION
**Pull request by yasharAyari**
### What was the problem?
We had a lot of unnecessary e2e test that makes our build process to slow.
### How did I fix it?
I replaced those e2e test with integration test 
### How to test it?
Run npm test
### Review checklist
- [ ] The PR solves #733 
- [ ] All new features are covered with e2e tests
- [ ] All new code follows best practices

### CODE REVIEW

File: src/components/voteDialog/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      port: 443,
+    },
+  };
+
+  store.dispatch(accountLoggedIn(realAccount));
+  store.dispatch({
+    data: peers,
+    type: actionTypes.activePeerSet,
+  });
+  const wrapper = mount(renderWithRouter(VoteDialog, store));
```
[the remark this answers is not in the document]
Comment (yasharAyari) [reply src/components/voteDialog/integration.test.js]: No, it shouldn't. because there is one component which will be test in each step and we don't need to mount it before each steps. if we do that , we will lose the last state.

File: src/components/voteDialog/integration.test.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  const clock = sinon.useFakeTimers({
+    toFake: ['setTimeout', 'clearTimeout', 'Date'],
+  });
+
+  step('user is login in', () => {
```
[the remark this answers is not in the document]
Comment (yasharAyari) [reply src/components/voteDialog/integration.test.js]: This is the initialize step . we don't need 'when' or 'then' in this step description.

File: src/components/voteDialog/integration.test.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    expect(store.getState().account).to.be.an('Object');
+    expect(store.getState().voting).to.be.an('Object');
+    expect(store.getState().peers).to.be.an('Object');
+  });
+
+  step('when he doesn\'t vote to any delegates confirm button should be disabled', () => {
```
[the remark this answers is not in the document]
Comment (yasharAyari) [reply src/components/voteDialog/integration.test.js]: I will replace he with 'user' to make it more general.

File: src/components/voteDialog/integration.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (slaweet) [reply src/components/voteDialog/integration.test.js]: It sounds like you confuse `Given` step with `beforeEach` block. What you are saying is true for `beforeEach` block, but not for `Given` step. Of course, you need to declare `let wrapper` outside of the step so that you can access it in other steps.

File: src/components/voteDialog/integration.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (slaweet) [reply src/components/voteDialog/integration.test.js]: `Given` is the initialization step.

See: https://martinfowler.com/bliki/GivenWhenThen.html
> The **given** part describes the state of the world before you begin the behavior you're specifying in this scenario. You can think of it as the pre-conditions to the test.
